crates/runtime/src/sources/solana_accounts_rpc.rs

```diff
@@ -0,0 +1,157 @@
+
+use async_trait::async_trait;
+use proto::tonic::Status;
+use solana_client::{
+    nonblocking::rpc_client::RpcClient,
+    rpc_config::{RpcAccountInfoConfig, RpcProgramAccountsConfig},
+};
+use solana_commitment_config::CommitmentConfig;
+use solana_pubkey::Pubkey;
+use tokio::sync::mpsc::Sender;
+use vixen_core::{Filters, GlobalFilters};
+use yellowstone_grpc_proto::geyser::{
+    subscribe_update::UpdateOneof, CommitmentLevel, SubscribeUpdate, SubscribeUpdateAccount,
+    SubscribeUpdateAccountInfo,
+};
+
+use crate::{config::YellowstoneConfig, sources::Source};
+
+/// A `Source` implementation for the Solana Accounts RPC API.
+#[derive(Debug, Default)]
+pub struct SolanaAccountsRpcSource {
+    config: Option<SolanaAccountsRpcConfig>,
+    filters: Option<Filters>,
+}
+
+impl SolanaAccountsRpcSource {
+    /// Create a new `SolanaAccountsRpcSource`.
+    #[must_use]
+    pub fn new() -> Self { Self::default() }
+
+    fn get_commitment_config(&self) -> CommitmentConfig {
+        match self.filters {
+            Some(Filters {
+                global_filters:
+                    GlobalFilters {
+                        commitment: Some(CommitmentLevel::Finalized),
+                    },
+                ..
+            }) => CommitmentConfig::finalized(),
+            Some(Filters {
+                global_filters:
+                    GlobalFilters {
+                        commitment: Some(CommitmentLevel::Processed),
+                    },
+                ..
+            }) => CommitmentConfig::processed(),
+            _ => CommitmentConfig::confirmed(),
+        }
+    }
+}
+
+/// The configuration for the Solana Accounts RPC source.
+#[derive(Debug, Clone)]
+pub struct SolanaAccountsRpcConfig {
+    /// The endpoint of the RPC server.
+    pub endpoint: String,
+    /// The timeout for the connection.
+    pub timeout: u64,
+}
+
+#[async_trait]
+impl Source for SolanaAccountsRpcSource {
+    async fn connect(
+        &self,
+        tx: Sender<Result<SubscribeUpdate, Status>>,
+    ) -> Result<(), crate::Error> {
+        // // We require that config and filters are set before connecting to the `Source`
+        let filters = self.filters.clone().ok_or(crate::Error::ConfigError)?;
+        let config = self.config.clone().ok_or(crate::Error::ConfigError)?;
+
+        let client = RpcClient::new_with_timeout_and_commitment(
+            config.endpoint,
+            Duration::from_secs(config.timeout),
+            self.get_commitment_config(),
+        );
+
+        for (filter_id, prefilter) in filters.parsers_filters {
+            if let Some(account_prefilter) = prefilter.account {
+                for program in account_prefilter.owners {
+                    let program_id = Pubkey::new_from_array(program.0);
+
+                    // TODO: Only get pubkey from get_program_accounts_with_config() using data_slice
+                    //  and then use get_multiple_accounts() for ~batching (process batches concurrently)
```

Comment:
Is this to get less data coming over the wire at a time? I've generally observed the key to getProgramAccounts is to not ever use a filter don't have the rpc server have to perform filter logic.